### ISSUE TITLE
Create SubUser - API response body fails on ReadAsStringAsync

### ISSUE THREAD

**Issue by andrewchilcott** [sendgrid/sendgrid-csharp#378]
#### Issue Summary

There are potentially two issues here (1) The response.Body.ReadAsStringAsync() is throwing an error  after sending a call to sg.client.subusers.post(requestBody:data) and (2) having deciphered the contents of the buffer in the response, it does not correspond to the response shown in the documentation.

#### Steps to Reproduce

1.  Send a call to create a subUser

      
```
        public async static Task<SubUser> createSubUserAsync(ApiKey key, SubUser subUser)
        {
            string data = @"{";
            data += "  'email': '" + subUser.Email + "',";
            data += "  'ips': [";
            for (int i = 0; i <= subUser.Ips.Count - 1; i++)
            {
                data += "'" + subUser.Ips[i] + "'";
                if (i != subUser.Ips.Count - 1)
                    data += ",";
            }
            data += "  ], ";
            data += "  'password': '" + subUser.Password + "',";
            data += "  'username': '" + subUser.UserName + "'";
            data += "}";
            Object json = JsonConvert.DeserializeObject<Object>(data);
            data = json.ToString();
            SubUser result = subUser;
            dynamic sg = new SendGrid.SendGridAPIClient(key.Key);
            dynamic response = await sg.client.subusers.post(requestBody: data);
            if (response.StatusCode.Equals(HttpStatusCode.Created))
            {
                //DEBUG CODE
                string strResult = response.Body.ReadAsStringAsync().Result;

                result = JsonConvert.DeserializeObject<SubUser>(response.Body.ReadAsStringAsync().Result);
                result.UserMessage = "Success - subUser created at " + DateTime.Now.ToString() + ".";
            }
            else
            {
                result.UserMessage = Shared.GenerateFailureMessage(response, "sub_user_create", subUser.UserName);
            }
            return result;
        }

```

2. The JsonConvert statement was throwing an error so I introduced strResult to find out what was in the response.Body. It still crashed on this line!

3. The response.StatusCode was Created. Checking on the portal, the Subuser had been created

4. I deciphered the contents of the buffer and found the following content:

```
{
"username":"MYUSERNAME",
"user_id":4260546,
"email":"MYEMAIL",
"signup_session_token":"",
"authorization_token":"",
"credit_allocation": {"type":"unlimited"}
}
```
The WebAPI version3 documentation shows

```
Create Subuser [POST]

**Request**

POST https://api.sendgrid.com/v3/subusers HTTP/1.1
Request Body

{
  "username": "John@example.com",
  "email": "John@example.com",
  "password": "johns_password",
  "ips": [
    "1.1.1.1",
    "2.2.2.2"
  ]
}

**Response**

HTTP/1.1 201
{
  "username": "John@example.com",
  "email": "John@example.com",
  "password": "johns_password",
  "ips": [
    "1.1.1.1",
    "2.2.2.2"
  ]
}

```

#### Technical details:

* sendgrid-csharp Version: 8.0.5.0 .NET Version: 4.5.2

**Comment by Jericho** [comment]
@andrewchilcott What is the exception you are getting? I don't have permission to access the 'subusers' resource so I can't test myself but based on your description, I think it sounds very similar to the issue you and I discussed in [this thread](https://github.com/sendgrid/sendgrid-csharp/issues/368)

**Comment by andrewchilcott** [comment]
@Jericho You're absolutely correct, there is a ContentType header of 'application/json; charset=utf8' in the response. I will use your workaround. Re: your comment about permissions, I worked out how to grant the subuser permissions - not documented!

This still leaves the issue that the response message contains different content to that in the documentation. What is the correct spec for the response message?

**Comment by andrewchilcott** [comment]
The same ContentType response problem occurs in sg.client.subusers.reputations.get(queryParams: queryParams);

**Comment by andrewchilcott** [comment]
Another call with the same ContentType response problem sg.client.subusers._(subUser.UserName).ips.put(requestBody: data);

**Comment by thinkingserious** [comment]
Thanks for all of the updates @andrewchilcott! I've added this to our backlog for a fix.

@Jericho, thanks for pointing him in the right direction!

**Comment by tony-ho** [comment]
I've created PR #786 which addresses the `charset=utf8` issue.

**Comment by childish-sambino** [comment]
Fixed by #786
